src/Components/Web.Extensions/src/wwwroot/headManager.js

```diff
@@ -0,0 +1,51 @@
+// Local helpers
+
+const getHeadElement = (tagName, id) => {
+    const elements = Array.from(document.getElementsByTagName(tagName));
+    return elements.find(e => e._blazorId === id);
+}
```

Comment:
This seems a bit odd because the name suggests it's only getting elements from `<head>`, but it actually gets them from anywhere. Also instead of implementing your own search, it might be worth using the built-in `querySelector` mechaism.

For example, when creating each element, define an extra attribute called `_blazor_head` with your GUID value. Then inside `getHeadElement` you can simply:

```js
return document.head.querySelector(`[_blazor_head='${id}']`);
```

And since the IDs are unique you don't need to care about tag names here.